shared/domains/chelonia/internals.js

```diff
@@ -1339,64 +1362,66 @@ const handleEvent = {
   },
   processSideEffects (message: GIMessage, state: Object) {
     const opT = message.opType()
-    if ([GIMessage.OP_ATOMIC, GIMessage.OP_ACTION_ENCRYPTED, GIMessage.OP_ACTION_UNENCRYPTED].includes(opT)) {
-      const contractID = message.contractID()
-      const manifestHash = message.manifest()
-      const hash = message.hash()
-      const id = message.id()
-      const signingKeyId = message.signingKeyId()
-
-      const callSideEffect = (field) => {
-        let v = field.valueOf()
-        let innerSigningKeyId: string | typeof undefined
-        if (isSignedData(v)) {
-          innerSigningKeyId = (v: any).signingKeyId
-          v = (v: any).valueOf()
-        }
+    if (![GIMessage.OP_ATOMIC, GIMessage.OP_ACTION_ENCRYPTED, GIMessage.OP_ACTION_UNENCRYPTED].includes(opT)) {
+      return
+    }
 
-        const { action, data, meta } = (v: any)
-        const mutation = {
-          data,
-          meta,
-          hash,
-          id,
-          contractID,
-          description: message.description(),
-          direction: message.direction(),
-          signingKeyId,
-          get signingContractID () {
-            return getContractIDfromKeyId(contractID, signingKeyId, state)
-          },
-          innerSigningKeyId,
-          get innerSigningContractID () {
-            return getContractIDfromKeyId(contractID, innerSigningKeyId, state)
-          }
-        }
-        return sbp('okTurtles.eventQueue/queueEvent', `sideEffect:${contractID}`, [`${manifestHash}/${action}/sideEffect`, mutation])
-      }
-      const msg = Object(message.message())
+    const contractID = message.contractID()
+    const manifestHash = message.manifest()
+    const hash = message.hash()
+    const id = message.id()
+    const signingKeyId = message.signingKeyId()
 
-      if (opT !== GIMessage.OP_ATOMIC) {
-        return callSideEffect(msg)
+    const callSideEffect = (field) => {
+      let v = field.valueOf()
+      let innerSigningKeyId: string | typeof undefined
+      if (isSignedData(v)) {
+        innerSigningKeyId = (v: any).signingKeyId
+        v = (v: any).valueOf()
       }
 
-      const reducer = (acc, [opT, opV]) => {
-        if ([GIMessage.OP_ACTION_ENCRYPTED, GIMessage.OP_ACTION_UNENCRYPTED].includes(opT)) {
-          acc.push(Object(opV))
+      const { action, data, meta } = (v: any)
+      const mutation = {
+        data,
+        meta,
+        hash,
+        id,
+        contractID,
+        description: message.description(),
+        direction: message.direction(),
+        signingKeyId,
+        get signingContractID () {
+          return getContractIDfromKeyId(contractID, signingKeyId, state)
+        },
+        innerSigningKeyId,
+        get innerSigningContractID () {
+          return getContractIDfromKeyId(contractID, innerSigningKeyId, state)
         }
-        return acc
       }
+      return Promise.resolve(sbp(`${manifestHash}/${action}/sideEffect`, mutation))
```

Comment:
Why is this wrapping the result in `Promise.resolve`? What if an error occurred? What's the point of later checking `.then` and `Promise.allSettled` if they're guaranteed to settled because of the `Promise.resolve`?

Or is that not how it works? If so please add a comment explaining because this is confusing...

Reply:
Agreed